lib/mix/lib/mix/scm/git.ex

```diff
@@ -201,18 +201,37 @@ defmodule Mix.SCM.Git do
   ## Helpers
 
   defp validate_git_options(opts) do
-    err =
-      "You should specify only one of branch, ref or tag, and only once. " <>
-        "Error on Git dependency: #{redact_uri(opts[:git])}"
+    opts
+    |> tap(&validate_single_refspec!/1)
+    |> tap(&validate_string_refspecs!/1)
```

Comment:
You can rewrite this to be something like

```elixir
# Returns [{key, value}] since it has to be one
[{_key, val}] = opts = validate_single_refspec(opts)

unless is_binary(val) do
  Mix.raise(...)
end

opts
```

And save a function and compress this quite a bit, thoughts?

Reply:
Seems reasonable, will update